bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -47,7 +47,12 @@ def run_plan(scope, plan_name, named_args = {})
       end
 
       begin
-        result = func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, params, true)
+        func.class.dispatcher.dispatchers[0].call_by_name_with_scope(scope, params, true)
+        # If the plan returns using a return statement then we never get here.
+        # We only get here if the plan finishes without executing any return
+        # statement and since we don't want to leak any implicit return value
+        # from the plan we simply set the result to nil here.
+        result = nil
```

Comment:
Ah I see what you were saying about making it harder to validate types. We'll have to catch the return for that.